site/src/docs/components/login/components.mdx

```diff
@@ -0,0 +1,205 @@
+| ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------- |
+| `onSuccess`                                                 | Required. The function is called when a response is handled successfully. The function is called with a User object. |
+| `onError`                                                   | Required. The function is called when a login fails. The function is called with an OidcClientError.                 |
+| `children`                                                  | Optional. Usually child elements to inform the user that a response is being processed.                              |
+| [Table 3: Properties of the LoginCallbackHandler component] |
+
+<PlaygroundPreview>
+
+```jsx
+const onSuccess = (user: User) => {
+  // handle success
+  // Usually redirect to a page
+};
+const onError = (error: OidcClientError) => {
+  // handle error
+};
+return (
+  <LoginProvider {...providerProperties}>
+    <LoginCallbackHandler onSuccess={onSuccess} onError={onError}>
+      <div>Logging in...</div>
+    </LoginCallbackHandler>
+  </LoginProvider>
+);
+```
+
+</PlaygroundPreview>
+
+### SessionEndedHandler
+
+This component listens to the session ended signal. It is emitted from the Session poller, if it receives a response indicating the user's session has ended.
+Then the component shows a dialog and forces the user to logout.
+
+### Properties
+
+| Property                                                   | Description                                                                                                       |
+| ---------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------- |
+| `content`                                                  | Required. Textual content with properties `title`, `text`, `buttonText` and `closeButtonLabelText`                |
+| `ids`                                                      | Optional. Ids for the dialog elements with properties `dialog`, `title`, and `text`. If omitted, ids are created. |
+| `dialogProps`                                              | Optional. Additional `DialogProps` passed to the Dialog                                                           |
+| [Table 4: Properties of the SessionEndedHandler component] |
+
+<PlaygroundPreview>
+
+```jsx
+<SessionEndedHandler
+  content={{
+    title: 'Session has ended!',
+    text: 'Your session on the server has ended. You will be logged out in this window too.',
+    buttonText: 'Logout',
+    closeButtonLabelText: 'Logout',
+  }}
+/>
+```
+
+</PlaygroundPreview>
+
+### WithAuthentication
+
+Component for rendering child components conditionally, depending on is user authenticated or not.
```

Comment:
`WithAuthenticatedUser` and `WithoutAuthenticatedUser` are wrappers and allow multiple children . That makes the `WithAuthenticatedUser` and `WithoutAuthenticatedUser` much more easier to use in most cases.

`WithAuthentication` accepts only components as props and it not so easy to use.

If needed, the `WithAuthentication` could be "unexported", but still kept as the renderer for the others.

Reply:
I see! What do you think, does `WithAuthentication` have some specific use cases where it's easier to use than it's counterparts? If there isn't any really, I would root for keeping `WithAuthentication` as an internal component that we use and don't export.

Reply:
When you want to use just one component for conditional rendering, it is clearer to use that one. And it has a (veeeeery minor) performance gain.

Reply:
Hmm it might be useful in displaying small pieces of UI that are located in the same place but are different for auth and non-auth users, like buttons. Maybe it can stay.